lib/blocks.php

```diff
@@ -57,7 +57,7 @@ function gutenberg_parse_blocks( $content ) {
 	 * If there are no blocks in the content, return a single block, rather
 	 * than wasting time trying to parse the string.
 	 */
-	if ( ! gutenberg_content_has_blocks( $content ) ) {
+	if ( ! has_blocks( $content ) ) {
```

Comment:
It has been brought up the potential for conflicts with function names. We might need to prefix with `wp_` here. But then the name can be seen as misleading "WordPress has blocks" (well, of course it does!).

Do we have similar functions in core that we should model after?

Reply:
In core, I can think of `has_shortcode` `has_excerpt` `has_category`  or `has_custom_logo`. 

Functions implemented with `wp_*` seem to be related to actions like `wp_{action}_{object}` like `wp_get_nav_menu` or `wp_get_attachment_{meta}`.

I would vote for `wp_` prefix just to make a difference from PHP native functions like `is_string` `is_readable` `file_exists`.

Anyway, I'm also thinking about this topic in #8340, and I would love to follow the decision here. In that case, I was thinking about a formula like `wp_{who}_has_{what}` for example:
*  `wp_post_has_block` where the $post parameter can be defaulted for `$post` global as many WordPress functions do
* `wp_content_has_block` Utility to quickly search for a block in a piece of content (maybe widgets content or excerpts.)

_Aside: And I don't even want to think about the fact that there may be a semantic difference between `block` and `block_type`._

Reply:
Chiming in with my own thoughts:

- Consistency is important. If there's already `has_shortcode`, `has_excerpt`, etc. the thought of introducing `wp_has_blocks` is worrying merely in that it breaks convention and therefore expectations of the developer.
   - On point of consistency, I'm a bit disappointed to see that `has_shortcode` accepts a string while `has_excerpt` accepts a post object. And then this function here accepts either / or 😬 
- Conflicts could occur, but the plugin handbook already [prescribes recommendations to avoid naming conflicts](https://developer.wordpress.org/plugins/the-basics/best-practices/#avoid-naming-collisions)
- Recent additions have been introduced without prefix, e.g. `register_rest_route`
- Whatever decision is made should probably be made and adopted universally, so that this discussion doesn't come up on every addition of a new core function 😆

Reply:
Yes, we should probably move this chat to the umbrella issue, as that was the point of it :D

Reply:
Aforementioned issue: #8352

Reply:
Given https://github.com/WordPress/gutenberg/issues/8352#issuecomment-413205212 and linked discussion in Slack meeting, I'm inclined to think this should remain `has_blocks` as proposed, for consistency with existing comparable WordPress function names.